### DISCUSSION TITLE
where do I define the bits to make the facade usable?

### DISCUSSION THREAD

**Discussion by lordisp** [Jeroen-G/laravel-packager#151] [Q&A] [answered]
usually I would expect to register the app service provider in `config\app.php` under `providers` and the class to `aliases`.

I've never done this on a package

Thanks!

**Comment by Jeroen-G** [comment, marked as answer]
For a facade from your own package?

Check this out: https://laravel.com/docs/8.x/packages#package-discovery

**Reply by lordisp** [reply]
I tried your `laravel-packager` as a starting point which includes that already. 

I've then created a simple method that returns a `foo` string. However I get a `Non-static method` exception message

It's registered in the sp:

```php
$this->app->singleton('azpimwebhook', function ($app) {
    return new AzPimWebhook;
});
```

```php
1) Tests\Feature\AzPimWebhookTest::facade_can_be_called                                       
Error: Non-static method Lordisp\AzPimWebhook\AzPimWebhook::test() cannot be called statically
```

not sure what I get wrong

**Reply by lordisp** [reply]
I've got it. Just had some typos. Thank you
